src/daft-sql/src/planner.rs

```diff
@@ -104,29 +136,26 @@ impl SQLPlanner {
         self.current_relation.as_ref()
     }
 
+    fn context_mut(&self) -> RefMut<'_, PlannerContext> {
+        self.context.as_ref().borrow_mut()
+    }
+
+    fn cte_map(&self) -> Ref<'_, HashMap<String, Relation>> {
+        Ref::map(self.context.borrow(), |i| &i.cte_map)
+    }
+
+    fn catalog(&self) -> Ref<'_, SQLCatalog> {
+        Ref::map(self.context.borrow(), |i| &i.catalog)
+    }
+
     /// Clears the current context used for planning a SQL query
     fn clear_context(&mut self) {
         self.current_relation = None;
         self.table_map.clear();
-        self.cte_map.clear();
+        self.context_mut().cte_map.clear();
     }
 
-    fn get_table_from_current_scope(&self, name: &str) -> Option<Relation> {
-        let table = self.table_map.get(name).cloned();
-        table
-            .or_else(|| self.cte_map.get(name).cloned())
-            .or_else(|| {
-                self.catalog
-                    .get_table(name)
-                    .map(|table| Relation::new(table.into(), name.to_string()))
-            })
-    }
```

Comment:
Removed this because I don't think it's actually correct to get columns from `cte_map` or `catalog` in most of the places this is called.